### ISSUE TITLE
[Ldap] Query attributes option

### ISSUE THREAD

**Issue by alborq** [symfony/symfony#19006]
Hi,
I have discover a little mistake. 
In method : `\Symfony\Component\Ldap\Adapter\ExtLdap\Query::execute`,
we can find call of : ldap_search

> Official php doc :
> `resource ldap_search ( resource $link_identifier , string $base_dn , string $filter [, array $attributes [, int $attrsonly [, int $sizelimit [, int $timelimit [, int $deref ]]]]] )`

arg nb 4 is $attributes, 

In `Query::execute`
call look like this : 

```
$this->search = @ldap_search(
    $con,
    $this->dn,
    $this->query,
    $this->options['filter'],
    $this->options['attrsOnly'],
    $this->options['maxItems'],
    $this->options['timeout'],
    $this->options['deref']
);
```

`$this->options['filter'],` is wrong.
must be change to : `$this->options['attributes'],`

and maybe ? follow this idea : 

> Official Php doc 
> Using this parameter is much more efficient than the default action (which is to return all attributes and their associated values). The use of this parameter should therefore be considered good practice

and change this attributs to required. 

ping @csarrazi

**Comment by csarrazi** [comment]
Actually, this is only a naming issue. If you take a look at the code, you'll see that the `ldap_search`'s `filter` parameter is called `query` in our case, and the `attributes` parameter is called `filter`.

Keep in mind that the Ldap component is not tied specifically to the PHP Ldap extension. We can change the parameters (or options in the config array), but once this is done, it will be final.

Oh, and currently, the current options actually reflect the content of the `find()` method from the previous implementation (see `LdapClient`).

**Comment by javiereguiluz** [comment]
Closing this old issue as "won't fix" because of the reasons given by Charles. Thanks!

**Comment by dontub** [comment]
I think the naming is confusing in regard to PHP's ldap functions (query => filter, filter => attributes). So it would help if it would be documented.

**Comment by csarrazi** [comment]
@dontub Please note that this bundle is an abstraction, the `query` method does not have the same signature as the PHP method, because the component aims to be an abstraction for not one, but multiple possible backends.

PHP's LDAP extension is only one of multiple possible implementations (Zend_LDAP could be another implementation, as well as a full user land library). Unfortunately I never managed to find the time to push this to the point where I want it.
